File: atcoder/denso2023_std/d.py
import itertools

n,m = map(int,input().split())
ab = [list(map(int,input().split())) for _ in range(m)]

for l in itertools.product([0,1,2], repeat=n):
    if all([l[a-1]!=l[b-1] for a,b in ab]):
        print("Yes")
        exit(0)

print("No")

File: atcoder/denso2023_std/d.rb
n,m=gets.split.map(&:to_i)
ab=(1..m).map{gets.split.map(&:to_i)}
puts [0,1,2].repeated_permutation(n).any?{|l|
    ab.all?{|a,b|l[a-1]!=l[b-1]}
} ? "Yes" : "No"
